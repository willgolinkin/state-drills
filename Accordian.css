* {
    box-sizing: border-box;
    /*normalize margins and padding*/
    margin: 0;
    padding: 0;
}
    
body, html {
    font-family: 'Karla', sans-serif; 
    color: black;
    text-align: center;
    /*for sticky footer*/
    height: 100%;
    margin: 0;
    padding: 0;
    width: 100%;
}
  
main {
    /*sets line spacing to desired range of 120-150%*/
    line-height: 150%;
}

.Accordian {
    list-style: none;
}