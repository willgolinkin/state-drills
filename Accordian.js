import React, { Component } from 'react';
import './Accordian.css';

export default class Accordian extends Component {
    
    static defaultProps = { sections: [] };
    //required to move harcoded 0 into state:
    constructor(props) {
        super(props)
        //state is an object and i have defined the key/value pair with current section index
        this.state = { activeSectionIndex: null, }
    };
    
    //handle state to update index on click
    handleSetActiveSection = (index) => {
        this.setState({ activeSectionIndex: index })
    }

    renderContent(section, idx, activeSectionIndex) {
        return (
            <li className='Accordian_item' key={idx}>
                <button type='button' onClick={() => this.handleSetActiveSection(idx)}>
                    {section.title}
                </button>
                {(activeSectionIndex === idx) && <p>{section.content}</p>}
            </li>
        )
    }

    render () {
        const { activeSectionIndex } = this.state
        const { sections } = this.props

        return (
                <ul className='Accordian'>
                    {sections.map((section, idx) => 
                        this.renderContent(section, idx, activeSectionIndex)
                    )}
                </ul>
        )
    }
}