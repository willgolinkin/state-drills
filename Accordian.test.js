import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordian from './Accordian';

const sectionsProp = [
    { title: 'First title',
        content: 'First Content'},
    { title: 'Second title',
        content: 'Second Content'},
    { title: 'Third title',
        content: 'Third Content'},
]

describe(`Accordion Component`, () => {
    it('renders empty given no Accordian without errors', () => {
      const wrapper = shallow(<Accordian />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  
    //The component renders no sections as active by default
    it('renders no sections as active by default', () => {
        const wrapper = shallow(<Accordian  sections={sectionsProp} />)
        expect (toJson(wrapper)).toMatchSnapshot()
    })

    it('opens any clicked section', () => {
        const wrapper = shallow(<Accordian sections={sectionsProp} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('only opens one section at a time', () => {
        const wrapper = shallow(<Accordian sections={sectionsProp} />)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})