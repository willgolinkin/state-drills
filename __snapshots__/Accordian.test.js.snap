// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component only opens one section at a time 1`] = `
<ul
  className="Accordian"
>
  <li
    className="Accordian_item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      First title
    </button>
  </li>
  <li
    className="Accordian_item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Second title
    </button>
  </li>
  <li
    className="Accordian_item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Third title
    </button>
    <p>
      Third Content
    </p>
  </li>
</ul>
`;

exports[`Accordion Component opens any clicked section 1`] = `
<ul
  className="Accordian"
>
  <li
    className="Accordian_item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      First title
    </button>
  </li>
  <li
    className="Accordian_item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Second title
    </button>
    <p>
      Second Content
    </p>
  </li>
  <li
    className="Accordian_item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Third title
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no Accordian without errors 1`] = `
<ul
  className="Accordian"
/>
`;

exports[`Accordion Component renders no sections as active by default 1`] = `
<ul
  className="Accordian"
>
  <li
    className="Accordian_item"
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      First title
    </button>
  </li>
  <li
    className="Accordian_item"
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Second title
    </button>
  </li>
  <li
    className="Accordian_item"
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Third title
    </button>
  </li>
</ul>
`;
